spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -363,24 +364,26 @@ protected AbstractEndpoint createEndpoint(MessageHandler handler, @SuppressWarni
 	protected AbstractEndpoint doCreateEndpoint(MessageHandler handler, MessageChannel inputChannel,
 			List<Annotation> annotations) {
 
-		Poller[] pollers = MessagingAnnotationUtils.resolveAttribute(annotations, "poller", Poller[].class);
+		Poller poller = MessagingAnnotationUtils.resolveAttribute(annotations, "poller", Poller.class);
+		boolean pollerProvided = poller != null && !ValueConstants.DEFAULT_NONE.equals(poller.value());
```

Comment:
Since the `ValueConstants.DEFAULT_NONE` is covered now in the `MessagingAnnotationUtils.hasValue()`, I guess we don't need it here any more.

Reply:
It is used [here](https://github.com/spring-projects/spring-integration/pull/3746/files#diff-4968e3d7ef5068b890a520af47f1fc8da4b9c369f7af56fbb05efbd5d07f070bR373) to ensure XOR on reactive|poller though.

Reply:
Yeh it is now simplified. Good code review @artembilan !!